internal/auth/credentials.go

```diff
@@ -16,14 +16,24 @@ func NewCredentialsStore() (credentials.Store, error) {
 	}
 
 	opts := credentials.StoreOptions{AllowPlaintextPut: false}
-	primaryStore, err := credentials.NewStore(configPath, opts)
+	notationStore, err := credentials.NewStore(configPath, opts)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create credential store from config file: %w", err)
 	}
 
-	fallbackStore, err := credentials.NewStoreFromDocker(opts)
+	dockerStore, err := credentials.NewStoreFromDocker(opts)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create credential store from docker config file: %w", err)
 	}
-	return credentials.NewStoreWithFallbacks(primaryStore, fallbackStore), nil
+
+	if !notationStore.IsAuthConfigured() && !dockerStore.IsAuthConfigured() {
+		// if both notation config and docker config do not have auth
+		// configuration, fallback to the platform-default credentials helper
+		// (if available)
+		osDefaultStore, ok := credentials.NewDefaultNativeStore()
+		if ok {
+			return credentials.NewStoreWithFallbacks(notationStore, dockerStore, osDefaultStore), nil
+		}
+	}
+	return credentials.NewStoreWithFallbacks(notationStore, dockerStore), nil
```

Comment:
If credential store is missing for notationStore, dockerStore, do we need to fallback to osDefaultStore?

Reply:
Yes, see line 29 - 37.

Reply:
```go
func (cfg *Config) IsAuthConfigured() bool {
	return cfg.credentialsStore != "" ||
		len(cfg.credentialHelpers) > 0 ||
		len(cfg.authsCache) > 0
}
```
If the `notationStore` and `dockerStore` only configure the `credentialHelpers` field, and `credentialsStore` is empty, `IsAuthConfigured` will return ture. In this case,  line 29 - 37 will be skipped and run line 38. Will line 38 also include osDefaultStore since the credentialsStore is currently empty?

Reply:
No, and I don't think we should fallback to `osDefaultStore` in this case.
IMHO, as long as there is any auth configuration in the config file, even if it's just `credHelpers`, we should honor that. Maybe the user just wanted to authenticate with a few specific registries and block others? 🤔

Reply:
Okey, it is reasonable.

Reply:
In this case, if user want to login, the error message will still be `credential store wasn't installed`, but the user may have installed the credential store?

Reply:
Is it possible that user installed Docker on linux without credential store, so the docker config `credsStore` field is empty. The user try to login with Notation, and Notation asked the user to install credential store. After the user installed the credential store, the docker config `credsStore` field is still empty, and user try to do Notation login again but still get the same error. I'm not sure that is it a valid case.

Reply:
> Is it possible that user installed Docker on linux without credential store, so the docker config credsStore field is empty. The user try to login with Notation, and Notation asked the user to install credential store. After the user installed the credential store, the docker config credsStore field is still empty, and user try to do Notation login again but still get the same error. I'm not sure that is it a valid case.

I think in this case the user should just configure the creds store in the notation config file or the docker config file. 😀